app/src/main/java/com/asfoundation/wallet/analytics/RakamAnalytics.kt

```diff
@@ -0,0 +1,111 @@
+package com.asfoundation.wallet.analytics
+
+import android.app.Application
+import android.content.Context
+import android.util.Log
+import com.asf.wallet.BuildConfig
+import com.asfoundation.wallet.identification.IdsRepository
+import com.asfoundation.wallet.logging.Logger
+import com.asfoundation.wallet.logging.RakamReceiver
+import io.rakam.api.Rakam
+import io.rakam.api.RakamClient
+import io.rakam.api.TrackingOptions
+import io.reactivex.Single
+import io.reactivex.schedulers.Schedulers
+import org.json.JSONException
+import org.json.JSONObject
+import java.net.MalformedURLException
+import java.net.URL
+
+class RakamAnalytics(private val context: Context, private val idsRepository: IdsRepository,
+                     private val logger: Logger) :
+    AnalyticsSetUp {
+  private val rakamClient = Rakam.getInstance()
+
+  companion object {
+    private val TAG = RakamAnalytics::class.java.simpleName
+  }
+
+  override fun setUserId(walletAddress: String) {
+    rakamClient.setUserId(walletAddress)
+  }
+
+  override fun setGamificationLevel(level: Int) {
+    val superProperties = rakamClient.superProperties ?: JSONObject()
+    try {
+      superProperties.put("user_level", level)
+    } catch (e: JSONException) {
+      e.printStackTrace()
+    }
+
+    rakamClient.superProperties = superProperties
+  }
+
+
+  fun start() {
+    Single.just(idsRepository.getAndroidId())
+        .flatMap { deviceId: String -> startRakam(deviceId) }
+        .flatMap { rakamClient: RakamClient ->
+          idsRepository.getInstallerPackage(BuildConfig.APPLICATION_ID)
+              .flatMap { installerPackage: String ->
+                Single.just(
+                    idsRepository.getGamificationLevel())
+                    .flatMap { level: Int ->
+                      Single.just(
+                          idsRepository.getActiveWalletAddress())
+                          .doOnSuccess { walletAddress: String ->
+                            setRakamSuperProperties(rakamClient,
+                                installerPackage,
+                                level, walletAddress)
```

Comment:
Improve codestyle by backspacing and applying codestyle again. Remove lambda variables type (level:Int ->  level)

Reply:
Done